src/store/slices/auth/index.ts

```diff
@@ -168,11 +168,19 @@ export const refreshTokensThunk = createAsyncThunk<void, void, { state: RootStat
     try {
       const currentAuthToken = await asyncStorageService.getItem(AsyncStorageKey.PhotosToken);
       if (!currentAuthToken) throw new Error('Auth token not found');
-      const { newToken, token } = await authService.refreshAuthToken(currentAuthToken);
+      const refreshed = await authService.refreshAuthToken(currentAuthToken);
 
+      if (!refreshed) {
```

Comment:
Not sure about this one, if we are unable to refresh the tokens (network request failed, 504 server error, or simply an error different than a 401) we don't logout the user since we assume the token can still be valid, otherwise this will logout us in cases such poor network conditions and being unable to communicate with the server. Anyways the old tokens are still "valid" based on their expirations.

So should we assume any API error should sign out the user or only the 401 ones?

Reply:
We should not, just 401.